test/serializers/meta_test.rb

```diff
@@ -67,11 +67,38 @@ def test_meta_is_not_used_on_arrays
         assert_equal expected, adapter.as_json
       end
 
+      def test_meta_is_present_on_arrays_with_root
+        serializer = ArraySerializer.new([@blog], meta: {total: 10}, meta_key: "haha_meta")
+        adapter = ActiveModel::Serializer::Adapter::Json.new(serializer, root: 'blog')
```

Comment:
so, you moved the `root: "blog"` form the constructor of the `ArraySerializer` to `ActiveModel::Serializer::Adapter::Json` because the way AMS is used in the controller, `root` is [sent only to the adapter, never to the serializer](https://github.com/rails-api/active_model_serializers/blob/1577969cb76309d1f48c68facc73e44c49489744/lib/action_controller/serialization.rb#L39)

Since it's basically wrong to pass `root` into `ArraySerializer`, maybe, not in this PR, we should make that raise an exception or issue a warning "Hey, serializer doesn't get root, the adapter does"